### PR TITLE
Merchant order show

### LINKED ISSUE #8: User Story 60, Merchant sees an order show page
```
As a merchant
When I visit an order show page from my dashboard
I see the customer's name and address
I only the items in the order that are being purchased from my inventory
I do not see any items in the order being purchased from other merchants
For each item, I see the following information:
- the name of the item, which is a link to my item's show page
- a small thumbnail of the item
- my price for the item
- the quantity the user wants to purchase
```

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #9

child of #79

depends on #31

### PR DESCRIPTION
**Pull request by teresa-m-knowles**
As a merchant
When I visit an order show page from my dashboard
I see the customer's name and address
I only the items in the order that are being purchased from my inventory
I do not see any items in the order being purchased from other merchants
For each item, I see the following information:
- the name of the item, which is a link to my item's show page
- a small thumbnail of the item
- my price for the item
- the quantity the user wants to purchase

Closes #8

### CODE REVIEW

File: spec/features/merchants/merchant_sees_order_spec.rb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      visit dashboard_path
+      click_link @order_1.id
+
+      expect(page).to have_link(@item_1.name)
+      expect(page).to have_link(@item_2.name)
+      expect(page).to have_link(@item_3.name)
+      expect(page).to have_link(@item_4.name)
+
+      #
+      # click_link(@item_1.name)
```
Comment (teresa-m-knowles) [inline spec/features/merchants/merchant_sees_order_spec.rb:65]: this is commented out because it was giving me a weird error that seems to be related to the items show page. Would love help here figuring out why though.

screen shot 2019-02-17 at 1 54 04 am

File: app/controllers/merchant/orders_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    @order = Order.find(params[:id])
+    @customer = @order.user
+    @items = Item.joins(:orders)
+                  .where(user: current_user.id)
+                  .group(:id)
+
+    @items_and_quantities = Hash.new
+
+    @items.each do |item|
+      @items_and_quantities[item] = item.order_items.where(order_id: @order.id).sum(:quantity)
```
Comment (csvlewis) [inline app/controllers/merchant/orders_controller.rb:12]: I could be wrong but this might be too much ActiveRecord being called in the controller. Would it be better to move some of these methods to the model level?

File: spec/features/merchants/merchant_sees_order_spec.rb (hunk 1 above)
Comment (timnallen) [reply spec/features/merchants/merchant_sees_order_spec.rb:65]: I bet because the order_items aren't fulfilled or something, so avg fulfillment time is being called on nil. I bet you could put an extra conditional on it that only runs it if at least one order item is fulfilled else it says not yet fulfilled or zero or something.

**Comment by timnallen** [comment]
This looks great

File: spec/features/merchants/merchant_sees_order_spec.rb (hunk 1 above)
Comment (csvlewis) [reply spec/features/merchants/merchant_sees_order_spec.rb:65]: I think it has something to do with when you try to go to one of the item show pages. Maybe it's because you're trying to do a calculation for fulfillment_time on an item that has never been fulfilled? You might need to put some logic in the item show page view to render some sort of default value if an item has not been fulfilled yet.

**Review by csvlewis — COMMENTED** [review]
I left some feedback on a couple things I noticed, I'd love to hear what other people in the group think. Overall it looks good though!

File: spec/features/merchants/merchant_sees_order_spec.rb (hunk 1 above)
Comment (csvlewis) [reply spec/features/merchants/merchant_sees_order_spec.rb:65]: I didn't see Tim's comment until after I had left mine but I think we had the same idea!

File: app/controllers/merchant/orders_controller.rb (hunk 2 above)
Comment (timnallen) [reply app/controllers/merchant/orders_controller.rb:12]: I think you may be right
